src/xunit.execution/Sdk/TypeUtility.cs

```diff
@@ -135,25 +135,51 @@ private static object TryConvertObject(object argumentValue, Type parameterType)
                 return argumentValue;
             }
 
+            // Implicit & explicit conversions to/from a type can be declared on either side of the relationship.
+            // We need to check both possibilities.
+            object convertedArgumentValue;
+
+            // Check for conversions declared on the parameter's type
+            if (TryPerformDefinedConversions(argumentValue, parameterType, out convertedArgumentValue))
+                return convertedArgumentValue;
+
+            // Check for conversions declared on the argument's type
+            if (TryPerformDefinedConversions(argumentValue, argumentValueType, out convertedArgumentValue))
+                return convertedArgumentValue;
+
+            // Can't convert object. We don't need to throw anything here, since MethodInfo.Invoke does
+            return argumentValue;
+        }
+
+        private static bool TryPerformDefinedConversions(
+            object argumentValue,
+            Type conversionDeclaringType,
+            out object convertedArgumentValue)
```

Comment:
Out is evil. There's no reason for out and bool return here, because a null value isn't a valid conversion, so it's a perfectly fine sentinel value for failed conversion.

Please re-write this to return the object, and above use null coalescing, like:

```
return PerformDefinedConversion(argumentValue, parameterType)
    ?? PerformDefinedConversion(argumentValue, argumentValueType)
    ?? argumentValue;
```